collector/lib/GetKernelObject.cpp

```diff
@@ -138,9 +140,10 @@ bool DownloadKernelObject(const std::string& hostname, const Json::Value& tls_co
   return false;
 }
 
-bool GetKernelObject(const std::string& hostname, const Json::Value& tls_config, const std::string& kernel_module, const std::string& module_path, bool verbose) {
-  std::string expected_path = kKernelModulesDir + "/" + kernel_module;
+bool GetKernelObject(const std::string& hostname, const Json::Value& tls_config, const DriverCandidate& candidate, bool verbose) {
+  std::string expected_path = candidate.getPath() + "/" + candidate.getName();
   std::string expected_path_compressed = expected_path + ".gz";
+  std::string module_path = candidate.isEbpf() ? SysdigService::kProbePath : SysdigService::kModulePath;
```

Comment:
I thought about this, but I believe the path where the drivers get extracted to are more related to the download mechanism than to the candidates themselves. Furthermore, all candidates of a given type would just have the path duplicated in them. Maybe when we start refactoring how the candidates are attempted to be used with an inspector and the fallback mechanism needed for BTF it would make more sense to pair them together, for now I feel like keeping them separated is the right move, WDYT?

Reply:
Yeah it's tricky. The extraction path, I'd argue, is less related to the download but related to the driver loading, and the expected path is only related to the download when the file does not currently exist. I do think this maybe needs some tweaking but can be in a follow up or when we adjust the loading logic.

Reply:
I've talked about this with @erthalion, I believe the end result here would be to have something like this:
```cpp
for (const auto& candidate : candidates) {
  auto inspector = CreateInspector(candidate);

  if (inspector) return inspector;
}
```

This would hide handling of the drivers completely from the main logic. In this scenario, the candidate only knows what the name and path to the driver are, there's no need for it to know whether it was downloaded, extracted, etc...

Obviously this also needs some pretty big refactoring , so there's not much point in doing that now, just want to make it a little clearer what my motivations are.